### PR TITLE
Move frontend from `src/web` to `src/web/app`

### PR DESCRIPTION
**Pull request by menghif**
## Issue This PR Addresses

Partial work for #2993

I'm breaking this issue into multiple PRs to make reviews easier.

## Type of Change

- [ ] **Bugfix**: Change which fixes an issue
- [ ] **New Feature**: Change which adds functionality
- [ ] **Documentation Update**: Change which improves documentation
- [ ] **UI**: Change which improves UI

## Description

This PR moves our frontend web app from `src/web` to `src/web/app`.

## Checklist

- [x] **Quality**: This PR builds and passes our npm test and works locally
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not (if applicable)
- [ ] **Documentation**: This PR includes updated/added documentation to user exposed functionality or configuration variables are added/changed or an explanation of why it does not(if applicable)

### CODE REVIEW

**Review by cindyledev — DISMISSED** [review]

**Review by DukeManh — APPROVED** [review]

**Comment by menghif** [comment]
I just noticed that the main banner image doesn't load on my machine. Does it work on your end? @cindyledev @DukeManh 
Also Vercel deployment is stuck and I don't know how to restart it.

**Comment by DukeManh** [comment]
@humphd Could you please update Vercel build from `src/web` to `src/web/app` too.

**Review by tcvan0707 — APPROVED** [review]

File: src/web/app/next.config.js [hunk 1]
```diff
@@ -37,12 +37,12 @@ const forwardToNext = (envVar) => {
 };
 
 // Try using .env in the root (legacy Telescope 1.0)
-const legacyEnvPath = path.join(__dirname, '../..', '.env');
+const legacyEnvPath = path.join(__dirname, '../../..', '.env');
```
Comment (humphd) [inline src/web/app/next.config.js:40]: Good job thinking of these...

**Review by humphd — APPROVED** [review]
Thanks for splitting this out, really helpful.

**Comment by menghif** [comment]
Rebased. @humphd Do we need to make changes to Vercel or can this be merged now?

**Review by DukeManh — DISMISSED** [review]

**Review by tcvan0707 — APPROVED** [review]

**Review by JerryHue — DISMISSED** [review]

**Review by cindyledev — APPROVED** [review]

**Comment by JerryHue** [comment]
Hmm... This is kind of problematic. Every PR that changes the front-end is going to conflict with this PR when it gets merged. There are currently 3 urgent PRs that change the front-end. Those PRs are #3282, #3281, #3211.

I suggest we merge those three tomorrow, and then immediately resolve the conflicts in this PR and immediately merge, so that other PRs that need to be started on the front-end can be done as soon as possible.

Otherwise, we would need to move this to `release 3.0-alpha` to avoid unnecessary conflicts and back-and-forth for several PRs.

**Comment by humphd** [comment]
What's our plan for landing this?  Git should be able to deal with file rename/moves automatically when rebasing.

**Review by JerryHue — DISMISSED** [review]

**Review by JerryHue — APPROVED** [review]

**Review by AmasiaNalbandian — APPROVED** [review]

**Pull request merged**